api/uniter/unit.go

```diff
@@ -740,39 +740,15 @@ func (u *Unit) AddStorage(constraints map[string][]params.StorageConstraints) er
 	return results.Combine()
 }
 
-// NetworkConfig requests network config information for the unit and the given
-// bindingName.
-func (u *Unit) NetworkConfig(bindingName string) ([]params.NetworkConfig, error) {
-	var results params.UnitNetworkConfigResults
-	args := params.UnitsNetworkConfig{
-		Args: []params.UnitNetworkConfig{{
-			BindingName: bindingName,
-			UnitTag:     u.tag.String(),
-		}},
-	}
-
-	err := u.st.facade.FacadeCall("NetworkConfig", args, &results)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	if len(results.Results) != 1 {
-		return nil, fmt.Errorf("expected 1 result, got %d", len(results.Results))
-	}
-
-	result := results.Results[0]
-	if result.Error != nil {
-		return nil, result.Error
-	}
-
-	return result.Config, nil
-}
-
-func (u *Unit) NetworkInfo(bindings []string) (map[string]params.NetworkInfoResult, error) {
+func (u *Unit) NetworkInfo(bindings []string, relationId *int) (map[string]params.NetworkInfoResult, error) {
 	var results params.NetworkInfoResults
 	args := params.NetworkInfoParams{
-		Unit:     u.tag.String(),
-		Bindings: bindings,
+		Unit:       u.tag.String(),
+		Bindings:   bindings,
+		RelationId: relationId,
+	}
+	if u.st.BestAPIVersion() < 7 {
```

Comment:
It may not be possible. It's just a sanity check to ensure that unexpected fields aren't sent to the v6 api, just in case we get strict about that at some point. The code here could very well be omitted, but I thought it reasonable to have as it also serves to highlight the v6 vs v7 difference.